app/services/LocationService.js

```diff
@@ -1,270 +1,295 @@
+    });
+
+    BackgroundGeolocation.on('http_authorization', () => {
+      console.log('[INFO] App needs to authorize the http requests');
+    });
+
+    BackgroundGeolocation.on('stop', () => {
+      // PUSH LOCAL NOTIFICATION HERE WARNING LOCATION HAS BEEN TERMINATED
+      console.log('[INFO] stop');
+    });
+
+    BackgroundGeolocation.on('stationary', () => {
+      // PUSH LOCAL NOTIFICATION HERE WARNING LOCATION HAS BEEN TERMINATED
+      console.log('[INFO] stationary');
+    });
+
+    BackgroundGeolocation.checkStatus(status => {
+      console.log(
+        '[INFO] BackgroundGeolocation service is running',
+        status.isRunning,
+      );
+      console.log(
+        '[INFO] BackgroundGeolocation services enabled',
+        status.locationServicesEnabled,
+      );
+      console.log(
+        '[INFO] BackgroundGeolocation auth status: ' + status.authorization,
+      );
+
+      BackgroundGeolocation.start(); //triggers start on start event
+
+      if (!status.locationServicesEnabled) {
+        // we need to set delay or otherwise alert may not be shown
+        setTimeout(
+          () =>
+            Alert.alert(
+              'Private Kit requires location services to be enabled',
+              'Would you like to open location settings?',
+              [
+                {
+                  text: 'Yes',
+                  onPress: () => BackgroundGeolocation.showLocationSettings(),
+                },
+                {
+                  text: 'No',
+                  onPress: () => console.log('No Pressed'),
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
+      } else if (!status.authorization) {
+        // we need to set delay or otherwise alert may not be shown
+        setTimeout(
+          () =>
+            Alert.alert(
+              'Private Kit requires access to location information',
+              'Would you like to open app settings?',
+              [
+                {
+                  text: 'Yes',
+                  onPress: () => BackgroundGeolocation.showAppSettings(),
+                },
+                {
+                  text: 'No',
+                  onPress: () => console.log('No Pressed'),
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
+      } else if (!status.isRunning) {
+      }
+    });
+
+    // you can also just start without checking for status
+    // BackgroundGeolocation.start();
+  }
+
+  static getPointCount() {
+    return lastPointCount;
+  }
+
+  static stop() {
+    // unregister all event listeners
+    BackgroundGeolocation.removeAllListeners();
+    BackgroundGeolocation.stop();
+    instanceCount -= 1;
+  }
+
+  static optOut(nav) {
+    BackgroundGeolocation.removeAllListeners();
+    BackgroundGeolocation.stop();
```

Comment:
This is the only code change right? just want to make sure that I am not missing any extra piece of code which has been added/removed.
Also, if possible please apply changes in such a way that code difference is minimized, right now it is hard to navigate across.

Reply:
Yes, that (BackgroundGeolocation.stop();) is the only code change I had made. The rest of the changes were due to the code formatter I used while making the change. Will be careful from next time and not use the code formatter. So, would you like me to close this pull request and create a new one or should we continue with this one?

Reply:
Hi there. Let's continue with this one. Please keep this PR open but push up a new commit that is only your change, and no code formatter changes. Thanks!